### PR TITLE
Move away from travis

### LINKED ISSUE #18: Update doc to transifex in a more frequent way like QGEP
As discussed with @m-kuhn 

https://github.com/QGEP/docs/commit/11fec125dd711b5874fe671eb9f18bbfbb183402#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34

### LINKED ISSUE #23: Enable Czech language
Hi, 
translating QWAT Doc is done. Please enable it on doc pages. 
Please enable Czech translation of plugin on Transifex.

Thanks, best regards, 
Jiri

### LINKED ISSUE #24: Translation is broken
See https://github.com/qwat/docs/pull/21 for more details

Do not use i18 files on transifex (these are wrong generated files )

image

### PR DESCRIPTION
**Pull request by ponceta**
On basis work of @m-kuhn,

Need cleaning of scripts/ci folder which is now useless

### CODE REVIEW

**Comment by ponceta** [comment]
Need to fix gettext and some Makefile functions to make it work smoothly. I'll try to get into sphinx and make these work on my computer before further changes.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: install requirements
+        run: |
+          pip install -r requirements.txt
+          sudo apt-get install -y gettext
+
+      - name: tx push / pull
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/move_away_from_travis' }}
```
Comment (ponceta) [inline .github/workflows/release.yml]: ping  @m-kuhn @ponceta

**Comment by ponceta** [comment]
image

image

Transifex has been pushed and pulled correctly. We still need to fix the syntax :

Existing : 
`PO-admin-guide--data_manager`
Newly created
`PO-i18n_pot_admin-guide_data_manager`

@m-kuhn it seems to be mandatory in the TX config file (even if it's a burden for maintenance)?

Or can we rename ressources in transifex to match a certain pattern?

**Comment by m-kuhn** [comment]
These files have been created by a test from me yesterday. I think we should be able to avoid listing the resources.

**Comment by ponceta** [comment]
@m-kuhn I've fixed on transifex side the ressources names (and removed the i18n files) so that they match with what is pushed from GitHub.

I see that translated chains or translated pages are not either pulled or build correctly what results in untranslated pages like : 
https://qwat.github.io/docs/fr/admin-guide/data_manager.html

**Comment by ponceta** [comment]
make transifex_pull seems to have difficulties on qwat_docs

image

**Comment by m-kuhn** [comment]
@ponceta are you taking this over or should I still have a look?

**Comment by ponceta** [comment]
@m-kuhn I could use a hand, I feel I miss something since I can retrieve, translate and build qgep_doc locally but not qwat_doc.

**Comment by ponceta** [comment]
image
index.html has not been updated since three days, maybe I miss something with the deploy process :S

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          TX_TOKEN=${{ secrets.TX_TOKEN }} make transifex_sync # push
+          TX_TOKEN=${{ secrets.TX_TOKEN }} make transifex_pull
+
+      - name: build
+        run: |
+          make html
+          touch build/html/.nojekyll
+
+      - name: Deploy 🚀
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/move_away_from_travis' }}
```
Comment (ponceta) [inline .github/workflows/release.yml]: Same remark as before

**Comment by m-kuhn** [comment]
@ponceta I will have a look end of this morning, ok for you?

**Comment by ponceta** [comment]
@m-kuhn this would be great! I really hope we can make this work! We will be then able to  focus on content and translations! :)

**Comment by m-kuhn** [comment]
Sorry I was blocked in the morning. Later today.

**Comment by m-kuhn** [comment]
> index.html has not been updated since three days, maybe I miss something with the deploy process :S

It looks like other files in the same place have been updated properly. Very likely this file simply did not change (so no reason to update).

File: .tx/config [hunk 3]
```diff
@@ -1,3 +1,245 @@
 [main]
 host = https://www.transifex.com
 
+[qwat-doc.admin-guide-data-manager]
+source_file = i18n/pot/admin-guide/data-manager.pot
```
Comment (m-kuhn) [inline .tx/config]: @ponceta this does not match the actual filename (`data_manager.pot`), was this done on purpose?

Comment (m-kuhn) [reply .tx/config]: FYI, I have a commit sitting here ready to completely generate this file, but only if you can confirm this should be `data_manager.pot`.

File: scripts/build-docs.sh [hunk 4]
```diff
@@ -1,45 +1,14 @@
-#!/usr/bin/env bash
+#!/bin/bash
```
Comment (m-kuhn) [inline scripts/build-docs.sh]: That was actually better the other way (more portable)

File: README.md [hunk 5]
```diff
@@ -1,24 +1,20 @@
...
+ * [Română](https://qwat.github.io/docs/ro/)
 
 ## Process
 
 The documentation is written in English and managed in this git repository.
-The latest version of the documentation is automatically built (on [Travis
-CI](https://travis-ci.org/qwat/docs)) and published on the links above.
+The latest version of the documentation is automatically built (using Github Actions) and published on the links above.
 
 Translation is done via transifex.
```
Comment (m-kuhn) [inline README.md:20]: Link to the qwat transifex project?
Keeping the entry barrier for contributors as small as possible.
```suggestion
Translation is done via transifex, [help translating by joining the team](https://www.transifex.com/qwat/qwat-doc/).
```

File: scripts/build-docs.sh (hunk 4 above)
Comment (ponceta) [reply scripts/build-docs.sh]:
```suggestion
#!/usr/bin/env bash
```
Let's revert (copy from QGEP -> adapt on QGEP ;)

File: .tx/config (hunk 3 above)
Comment (ponceta) [reply .tx/config]: This is indeed not the right syntax if we want to map data_manager.rst file. I let you check this and the other files (.pot and po) below if you have a commit ready.

data-manager -> data_manager

File: .tx/config [hunk 6]
```diff
@@ -1,3 +1,245 @@
...
+source_file = i18n/pot/contributor-guide/index.pot
+source_lang = en
+trans.de = i18n/de/demo-guide/index.po
+trans.fr = i18n/fr/demo-guide/index.po
+trans.it = i18n/it/demo-guide/index.po
+trans.ro = i18n/ro/demo-guide/index.po
+type = PO
+
+[qwat-doc.developer-guide-index]
+source_file = i18n/pot/developer/guide-index.pot
```
Comment (ponceta) [inline .tx/config]:
```suggestion
source_file = i18n/pot/developer-guide/index.pot
```

File: .tx/config [hunk 7]
```diff
@@ -1,3 +1,245 @@
...
+source_lang = en
+trans.de = i18n/de/developer-guide/index.po
+trans.fr = i18n/fr/developer-guide/index.po
+trans.it = i18n/it/developer-guide/index.po
+trans.ro = i18n/ro/developer-guide/index.po
+type = PO
+
+[qwat-doc.developer-guide-local-customizations]
+source_file = i18n/pot/developer-guide/local-customizations.pot
```
Comment (ponceta) [inline .tx/config]:
```suggestion
source_file = i18n/pot/developer-guide/local_customizations.pot
```

File: .tx/config [hunk 8]
```diff
@@ -1,3 +1,245 @@
...
+source_lang = en
+trans.de = i18n/de/developer-guide/local-customizations.po
+trans.fr = i18n/fr/developer-guide/local-customizations.po
+trans.it = i18n/it/developer-guide/local-customizations.po
+trans.ro = i18n/ro/developer-guide/local-customizations.po
+type = PO
+
+[qwat-doc.developer-guide-qwat-dev]
+source_file = i18n/pot/developer-guide/qwat-dev.pot
```
Comment (ponceta) [inline .tx/config]:
```suggestion
source_file = i18n/pot/developer-guide/qwat_dev.pot
```

File: .tx/config [hunk 9]
```diff
@@ -1,3 +1,245 @@
...
+source_file = i18n/pot/faq/index.pot
+source_lang = en
+trans.de = i18n/de/faq/index.po
+trans.fr = i18n/fr/faq/index.po
+trans.it = i18n/it/faq/index.po
+trans.ro = i18n/ro/faq/index.po
+type = PO
+
+[qwat-doc.features-guide-index]
+source_file = i18n/pot/features/guide-index.pot
```
Comment (ponceta) [inline .tx/config]:
```suggestion
source_file = i18n/pot/features-guide/index.pot
```

File: .tx/config [hunk 10]
```diff
@@ -1,3 +1,245 @@
...
+source_file = i18n/pot/index.pot
+source_lang = en
+trans.de = i18n/de/index.po
+trans.fr = i18n/fr/index.po
+trans.it = i18n/it/index.po
+trans.ro = i18n/ro/index.po
+type = PO
+
+[qwat-doc.installation-guide-database-initialization]
+source_file = i18n/pot/installation-guide/database-initialization.pot
```
Comment (ponceta) [inline .tx/config]: This is actually OK, but we could change the rst file name with an underscore if we want to be systematic with files namings

Comment (m-kuhn) [reply .tx/config]: That is a good idea, I'm just not sure if transifex will transfer all the existing translations or if that needs manual work.

**Comment by ponceta** [comment]
@m-kuhn this seems fine in the build and actions process but actually still does not the job of applying the translations to the files :
https://qwat.github.io/docs/fr/features-guide/index.html as an example.

**Comment by m-kuhn** [comment]
yep, I am looking at this

**Comment by ponceta** [comment]
i18n_pot_* ressources have been pushed again toward transifex, maybe that's where the translation is stuck :S
ex :
i18n_pot_admin-guide_data_manager

**Comment by m-kuhn** [comment]
@ponceta this looks good to me now:

https://qwat.github.io/docs/fr/user-guide/update.html

**Comment by m-kuhn** [comment]
The current approach is to store the complete .tx/config file in the source. It will be automatically generated through github actions, so we should always have an up to date version. This is not really a perfect workflow but should be good enough for the moment.

Also note that we have other hard to discover parts of the system. E.g. on transifex there is a high percentage of Czech (cs) translation, however each language needs to be added to the Makefile. Cs is added now.

**Comment by ponceta** [comment]
Aie, @m-kuhn I should have warned you, QWAT doc is partly still not available in english...

It is mostly the case for the user guide:
https://qwat.github.io/docs/fr/user-guide/update.html

But if you look at a translated en->fr page the translation workflow does not work
https://qwat.github.io/docs/fr/features-guide/index.html

Same for Czech :

https://qwat.github.io/docs/cs/features-guide/index.html

(although translations are available on transifex and correctly downloaded if you try it locally)

None of the translated html pages has been pushed to gh-page : (only doc-tree refresh)

image

**Comment by m-kuhn** [comment]
Thanks for the feedback and sorry for missing that.
I can confirm translated files (.po) get pulled from transifex, get compiled (.mo) but are not integrated in the sphinx-build.

**Comment by ponceta** [comment]
Is it possible that the fact that we have several index.po files could make trouble?

image

I was not able to check the *.mo files content to see if all translations are well compiled.

**Comment by m-kuhn** [comment]
Confirmed, this is exactly the problem.
We need to setup the equal directory structure

File: README.md [hunk 11]
```diff
@@ -1,24 +1,20 @@
...
- * [English](https://qwat.github.io/docs/master/en/html)
- * [Deutsch](https://qwat.github.io/docs/master/de/html)
- * [Français](https://qwat.github.io/docs/master/fr/html)
- * [Italiano](https://qwat.github.io/docs/master/it/html)
- * [Română](https://qwat.github.io/docs/master/ro/html)
+ * [English](https://qwat.github.io/docs/en/)
+ * [Deutsch](https://qwat.github.io/docs/de/)
+ * [Français](https://qwat.github.io/docs/fr/)
+ * [Italiano](https://qwat.github.io/docs/it/)
+ * [Română](https://qwat.github.io/docs/ro/)
```
Comment (m-kuhn) [inline README.md:12]:
```suggestion
 * [Română](https://qwat.github.io/docs/ro/)
 * [Czech](https://qwat.github.io/docs/cs/)
```

File: scripts/create_tx_conf.py [hunk 12]
```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+
+# Creates a transifex configuration from pot files that are found in i18n/pot
```
Comment (m-kuhn) [inline scripts/create_tx_conf.py:3]:
```suggestion
# Creates a transifex configuration from pot files that are found in i18n/pot
# Will be automatically run and committed via github actions
```

**Review by m-kuhn — APPROVED** [review]
@ponceta I left som final suggestions for your consideration.

File: .github/workflows/release.yml [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+        with:
+          commit_message: Update .tx/config
+
+      - name: build
+        run: |
+          make html
+          touch build/html/.nojekyll
+
+      - name: Deploy 🚀
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/maste' }}
```
Comment (ponceta) [inline .github/workflows/release.yml]:
```suggestion
        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/master' }}
```
Typo : master

**Comment by ponceta** [comment]
Thank you @m-kuhn ! 👍

**Pull request merged**